instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/HttpServerRequestMetrics.java

```diff
@@ -0,0 +1,132 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class HttpServerRequestMetrics implements RequestMetrics {
+
+  private static final ContextKey<State> HTTP_SERVER_REQUEST_METRICS_STATE =
+      ContextKey.named("http-server-request-metrics-state");
+
+  private static final Logger logger = LoggerFactory.getLogger(HttpServerRequestMetrics.class);
+
+  public static RequestMetricsFactory factory() {
+    return HttpServerRequestMetrics::new;
+  }
+
+  private final LongUpDownCounter activeRequests;
+  private final DoubleValueRecorder duration;
+
+  private HttpServerRequestMetrics(Meter meter) {
+    activeRequests =
+        meter
+            .longUpDownCounterBuilder("http.server.active_requests")
+            .setUnit("requests")
+            .setDescription("The number of concurrent HTTP requests that are currently in-flight")
+            .build();
+
+    duration =
+        meter
+            .doubleValueRecorderBuilder("http.server.duration")
+            .setUnit("milliseconds")
+            .setDescription("The duration of the inbound HTTP request")
+            .build();
+  }
+
+  @Override
+  public Context start(Context context, Attributes requestAttributes) {
+    long startTimeNanos = System.nanoTime();
+    Labels activeRequestLabels = activeRequestLabels(requestAttributes);
+    Labels durationLabels = durationLabels(requestAttributes);
+    activeRequests.add(1, activeRequestLabels);
+
+    return context.with(
+        HTTP_SERVER_REQUEST_METRICS_STATE,
+        new AutoValue_HttpServerRequestMetrics_State(
+            activeRequestLabels, durationLabels, startTimeNanos));
+  }
+
+  @Override
+  public void end(Context context, Attributes responseAttributes) {
+    State state = context.get(HTTP_SERVER_REQUEST_METRICS_STATE);
+    if (state == null) {
+      logger.debug(
+          "No state present when ending context {}. Cannot reset HTTP request metrics.", context);
+      return;
+    }
+    activeRequests.add(-1, state.activeRequestLabels());
+    duration.record(System.nanoTime() - state.startTimeNanos(), state.durationLabels());
+  }
+
+  private static Labels activeRequestLabels(Attributes attributes) {
+    LabelsBuilder labels = Labels.builder();
+    attributes.forEach(
+        (key, value) -> {
+          if (key.getType() != AttributeType.STRING) {
+            return;
+          }
+          switch (key.getKey()) {
+            case "http.method":
+            case "http.host":
+            case "http.scheme":
+            case "http.flavor":
+            case "http.server_name":
+              labels.put(key.getKey(), (String) value);
+              break;
+            default:
+              // fall through
+          }
+        });
+    return labels.build();
+  }
+
+  private static Labels durationLabels(Attributes attributes) {
+    LabelsBuilder labels = Labels.builder();
+    attributes.forEach(
+        (key, value) -> {
+          if (key.getType() != AttributeType.STRING) {
```

Comment:
`http.status_code` and `net.host.port` are longs, not strings.

Reply:
Hmm - status code isn't ready anyways when I compute these labels at the beginning of the request. I guess depending on the metric, some need to be computed at beginning (active requests) and some at end (duration). But since there's no text in the spec on that yet I'll keep it as is for now.